Calorimetry/python/jetResolution/histPool.py

```diff
@@ -0,0 +1,151 @@
+"""Helper class which contains all histograms."""
+import ROOT
+import math
+import array
+from helperFunctions import randomString
+from helperFunctions import calculateRMS90
+from helperFunctions import S_OK, S_ERROR
+from collections import namedtuple
+
+
+VectorTuple = namedtuple('VectorTuple', "quark1 quark2 totalTrueVisible "
+                         "totalTrueInvisible totalPFO genJets recoJets")
+
+
+class histPool(object):
+  """Helper class which contains all histograms."""
+
+  cosThetaRange = [0.0, 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 0.925,
+                   0.95, 0.975, 1.0]
+  histInfo = [['TH1F', 'cosTheta', 'cosTheta; cos(#theta); Entries',
+              50, 0, 1.0],
+              ['TH1F', 'true_totalE', '; total true E [GeV]; Entries',
+              400, 200, 300],
+              ['TH1F', 'gen_totalE', '; gen di-jet E [GeV]; Entries',
+              250, 0, 250],
+              ['TH1F', 'reco_totalE', '; reco di-jet E [GeV]; Entries',
+              250, 0, 250],
+              ['TH1F', 'gen_InvMass', '; gen inv.mass [GeV]; Entries',
+              250, 0, 250],
+              ['TH1F', 'reco_InvMass', '; reco inv.mass [GeV]; Entries',
+              250, 0, 250],
+              ['TH1F', 'JER', 'cosTheta; cos(#theta); JER [%]',
+              len(cosThetaRange) - 1, cosThetaRange]
+  ]
+
+  def __init__(self):
+    """Initialize."""
+    self.histDict2D = {}
+
+  def initNewSetOfHists(self, setName, histInformation):
+    """Initialize new set of histograms."""
+    if setName in self.histDict2D:
+      return True
+    tmpDict = {}
+    for iHistInfo in histInformation:
+      if iHistInfo[0] == 'TH1F':
+        if len(iHistInfo) == 5:
+          tmpArr = array.array('d', iHistInfo[4])
+          tmpHist = ROOT.TH1F(randomString(10), iHistInfo[2], iHistInfo[3],
+                              tmpArr)
+        if len(iHistInfo) == 6:
+          tmpHist = ROOT.TH1F(randomString(10), iHistInfo[2], iHistInfo[3],
+                              iHistInfo[4], iHistInfo[5])
+        tmpDict[iHistInfo[1]] = tmpHist
+    self.histDict2D[setName] = tmpDict
+
+  def createJERFitHistInfo(self):
+    """???"""
+    outHistInfo = []
+    histInfoTemplate = ['TH1F', 'cosThetaBin', '; reco + true neutrino energy [GeV]; Entries',
+                        10000, 0., 500.]
```

Comment:
for CLIC the upper limit might need larger numbers, for sure larger than 1500., I even observed very high tail events up to 3000.